compiler/src/iree/compiler/Dialect/Stream/Transforms/test/elide_transfer_ops.mlir

```diff
@@ -0,0 +1,98 @@
+// RUN: iree-opt --split-input-file --iree-stream-elide-transfer-ops %s | FileCheck %s
+
+module @test attributes {
+  stream.topology = #hal.device.topology<links = [
+    (@dev_a -> @dev_b = {unified_memory = true}),
+    (@dev_b -> @dev_a = {unified_memory = true}),
+    (@dev_a -> @dev_c = {})
+  ]>
+} {
+
```

Comment:
split the tests into their own chunks with `// -----`

It'll mean duplicating the topology, but that keeps tests hermetic and makes it clear what matters. It _shouldn't_ be the same for all the tests, as then it's unclear to future maintainers what is actually meaningful (if a test only uses dev_a and dev_b, the dev_c link is noise), etc).